### ISSUE TITLE
Implement Gradient Clipping

### ISSUE THREAD

**Issue by RAGoody** [RAGoody/neuralNetModeler#5]
As a trainer, I am seeing the model explode with too-large floats due to a lack of gradient clipping.

UAC
- using larger learning rates should not lead to math overruns.
